compose/config/validation.py

```diff
@@ -151,18 +151,29 @@ def handle_error_for_schema_with_id(error, service_name):
             VALID_NAME_CHARS)
 
     if schema_id == '#/definitions/constraints':
+        # Build context could in 'build' or 'build.context' and dockerfile could be
+        # in 'dockerfile' or 'build.dockerfile'
```

Comment:
With my proposed change to have `process_service()` always return build as a dict, I don't think we'll need to worry about this here either. We can assume a consistent representation.